app/com/linkedin/drelephant/spark/fetchers/SparkRestClient.scala

```diff
@@ -76,17 +78,20 @@ class SparkRestClient(sparkConf: SparkConf) {
     .property(ClientProperties.READ_TIMEOUT, READ_TIMEOUT).target(historyServerUri).path(API_V1_MOUNT_PATH)
 
 
-  def fetchData(appId: String, fetchLogs: Boolean = false, fetchFailedTasks: Boolean = true)(
+  def fetchData(appId: String, fetchLogs: Boolean = false, retriesRemaining: Int = Int.MaxValue, fetchFailedTasks: Boolean = true)(
```

Comment:
Why the default value of retriesRemaining is Int.MaxValue ?

Reply:
This is done for the TEST CASES where retires remaining is not provided, it would prevent those sort of test failing due to this condition.

Reply:
But can we put some default value (instead of Integer.MaxValue) . Integer.MaxValue looks misleading for retries remaining variable